webapp/src/components/calendar/fullCalendar.tsx

```diff
@@ -0,0 +1,206 @@
+}
+
+function createDatePropertyFromCalendarDates(start: Date, end: Date) : DateProperty {
+    // save as noon local, expected from the date picker
+    start.setHours(12)
+    const dateFrom = start.getTime() - timeZoneOffset(start.getTime())
+    end.setHours(12)
+    const dateTo = end.getTime() - timeZoneOffset(end.getTime()) - oneDay // subtract one day. Calendar is date exclusive
+
+    const dateProperty : DateProperty = {from: dateFrom}
+    if (dateTo !== dateFrom) {
+        dateProperty.to = dateTo
+    }
+    return dateProperty
+}
+
+const timeZoneOffset = (date: number): number => {
+    return new Date(date).getTimezoneOffset() * 60 * 1000
+}
+
+const CalendarFullView = (props: Props): JSX.Element|null => {
+    const intl = useIntl()
+    const {board, cards, activeView, dateDisplayProperty, readonly} = props
+    const visiblePropertyTemplates = board.fields.cardProperties.filter((template: IPropertyTemplate) => activeView.fields.visiblePropertyIds.includes(template.id))
+    const isSelectable = !readonly
+
+    let {initialDate} = props
+    if (!initialDate) {
+        initialDate = new Date()
+    }
+
+    const isEditable = () : boolean => {
+        if (readonly || !dateDisplayProperty || (dateDisplayProperty.type === 'createdTime' || dateDisplayProperty.type === 'updatedTime')) {
+            return false
+        }
+        return true
+    }
+
+    const myEventsList = cards.flatMap((card): EventInput[] => {
+        let dateFrom = new Date(card.createAt || 0)
+        let dateTo = new Date(card.createAt || 0)
+        if (dateDisplayProperty && dateDisplayProperty?.type === 'updatedTime') {
+            dateFrom = new Date(card.updateAt || 0)
+            dateTo = new Date(card.updateAt || 0)
+        } else if (dateDisplayProperty && dateDisplayProperty?.type !== 'createdTime') {
+            const dateProperty = createDatePropertyFromString(card.fields.properties[dateDisplayProperty.id || ''] as string)
+            if (!dateProperty.from) {
+                return []
+            }
+
+            // date properties are stored as 12 pm UTC, convert to 12 am (00) UTC for calendar
+            dateFrom = dateProperty.from ? new Date(dateProperty.from + (dateProperty.includeTime ? 0 : timeZoneOffset(dateProperty.from))) : new Date()
+            dateFrom.setHours(0, 0, 0, 0)
+            const dateToNumber = dateProperty.to ? dateProperty.to + (dateProperty.includeTime ? 0 : timeZoneOffset(dateProperty.to)) : dateFrom.getTime()
+            dateTo = new Date(dateToNumber + oneDay) // Add one day.
+            dateTo.setHours(0, 0, 0, 0)
+        }
+        return [{
+            id: card.id,
+            title: card.title,
+            extendedProps: {icon: card.fields.icon},
+            properties: card.fields.properties,
+            allDay: true,
+            start: dateFrom,
+            end: dateTo,
+        }]
+    })
+
+    const renderEventContent = (eventProps: EventContentArg): JSX.Element|null => {
```

Comment:
Maybe another component (maybe not)

Reply:
I was going to make a separate component, but that meant we had to set the board, card and visible properties for each card in order for that information to be available to the separate component. Seemed like overkill, so I left it inside this component.

Reply:
That sounds good to me :)